android-components/components/concept/engine/src/main/java/mozilla/components/concept/engine/EngineSession.kt

```diff
@@ -780,6 +780,13 @@ abstract class EngineSession(
      */
     abstract fun toggleDesktopMode(enable: Boolean, reload: Boolean = false)
 
+    /**
+     * Checks if there is a rule for handling a cookie banner for the current website in the session.
+     *
+     * @return true if there is a cookie banner handling rule, false otherwise.
+     */
+    abstract fun hasCookieBannerRuleForSession(onSuccess: (Boolean?) -> Unit, onError: (Throwable) -> Unit)
```

Comment:
Nit: We should document both parameters, specially what `onSuccess(null)` indicates.

Reply:
Just a head up, as this is a generic API we don't propagate specifics implementation details about internal behaviours of the engines (`GeckoEngine` or `SystemEngine`), if the `GeckoEngine` provide `null` values we don't expose it as part of public APIs unless the same behaviour could apply for both engines.